aten/src/ATen/native/TensorIterator.cpp

```diff
@@ -47,9 +47,9 @@ void TensorIterator::reorder_dimensions(const TensorIteratorConfig& config) {
       }
       if (stride0 == 0 || stride1 == 0) {
         continue;
-      } else if (stride0 <= stride1) {
+      } else if (stride0 < stride1) {
         return -1;
-      } else {
+      } else  if (stride0 > stride1) {
         return 1;
       }
```

Comment:
so we treat equal dims as ambigious? shouldn't we do something like "preserver the original order", i.e. compare dim0 and dim1 themselves? Maybe that's what the insertion sort below does, but it's hard for me to mentally unravel it

Reply:
This is a very good question! Added a "flies in the ointment" section to the description. When strides are equal, we don't swap corresponding dimension, so we tend to favor identity (or closer-to-identity) permutation in this case.